docs/src/modules/utils/generateMarkdown.js

```diff
@@ -273,6 +273,14 @@ function generateProps(reactAPI) {
     return textProps;
   }, text);
 
+  let refHint = 'The `ref` is forwarded to the root element.';
+  if (reactAPI.forwardsRefTo == null) {
+    refHint = 'The component cannot hold a ref.';
+  } else if (reactAPI.forwardsRefTo === 'React.Component') {
+    refHint = 'The `ref` is attached to a component class.';
```

Comment:
TouchRipple is the only candidate for that (allows imperative handle). The other components will be covered by #15170.